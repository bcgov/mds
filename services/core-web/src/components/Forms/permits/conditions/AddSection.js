import React, { useState } from "react";
import PropTypes from "prop-types";
import AddButton from "@/components/common/AddButton";
import Section from "@/components/Forms/permits/conditions/Section";

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
};

const defaultProps = {};

const AddSection = (props) => {
  const [isEditing, setIsEditing] = useState(false);
  return (
    <>
      {!isEditing && (
        <AddButton type="secondary" onClick={() => setIsEditing(true)}>
          Add Sub-Section
        </AddButton>
      )}
      {isEditing && (
        <Section
          new
          handleSubmit={(values) => {
            setIsEditing(false);
            props.handleSubmit(values);
          }}
          initialValues={props.initialValues}
        />
      )}
    </>
  );
};

AddSection.propTypes = propTypes;
AddSection.defaultProps = defaultProps;

export default AddSection;
