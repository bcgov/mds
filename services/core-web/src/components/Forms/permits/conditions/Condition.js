import React from "react";
import PropTypes from "prop-types";
import SubCondition from "@/components/Forms/permits/conditions/SubCondition";
import Section from "@/components/Forms/permits/conditions/Section";
import ListItem from "@/components/Forms/permits/conditions/ListItem";

const propTypes = {
  condition: PropTypes.objectOf(PropTypes.any).isRequired,
};

const ConditionComponent = (props) =>
  ({
    SEC: <Section {...props} />,
    CON: <SubCondition {...props} />,
    LIS: <ListItem {...props} />,
  }[props.condition.condition_type_code]);

const Condition = (props) => <ConditionComponent {...props} />;

Condition.propTypes = propTypes;

export default Condition;
