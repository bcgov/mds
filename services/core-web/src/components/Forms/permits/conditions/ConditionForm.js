import React from "react";
import PropTypes from "prop-types";
import { Field, reduxForm } from "redux-form";
import { Form } from "@ant-design/compatible";
import "@ant-design/compatible/assets/index.css";
import { Button, Popconfirm, Row, Col } from "antd";
import { required } from "@common/utils/Validate";
import { resetForm } from "@common/utils/helpers";
import * as FORM from "@/constants/forms";
import { renderConfig } from "@/components/common/config";

const propTypes = {
  onCancel: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  layer: PropTypes.number.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
};

const placeHolderText = (type) =>
  ({
    SEC: "Sub-Section Name",
    CON: "Type a condition",
    LIS: "Type a list item",
  }[type]);

export const ConditionForm = (props) => {
  const formSpan = 21 - props.layer;
  return (
    <Row gutter={[16, 24]}>
      <Col span={props.layer} />
      <Col span={formSpan}>
        <Form onSubmit={props.handleSubmit}>
          <Field
            id="condition"
            name="condition"
            placeholder={placeHolderText(props.initialValues.condition_type_code)}
            required
            component={
              props.initialValues.condition_type_code === "SEC"
                ? renderConfig.FIELD
                : renderConfig.AUTO_SIZE_FIELD
            }
            validate={[required]}
          />
          <div className="right center-mobile">
            <Popconfirm
              placement="topRight"
              title="Are you sure you want to cancel?"
              onConfirm={props.onCancel}
              okText="Yes"
              cancelText="No"
            >
              <Button className="full-mobile" type="secondary">
                Cancel
              </Button>
            </Popconfirm>
            <Button
              className="full-mobile"
              type="primary"
              htmlType="submit"
              loading={props.submitting}
            >
              Save
            </Button>
          </div>
        </Form>
      </Col>
      <Col span={3} />
    </Row>
  );
};

ConditionForm.propTypes = propTypes;

export default reduxForm({
  form: FORM.CONDITION_SECTION,
  onSubmitSuccess: resetForm(FORM.CONDITION_SECTION),
})(ConditionForm);
