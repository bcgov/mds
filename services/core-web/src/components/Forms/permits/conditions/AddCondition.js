import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Menu, Dropdown } from "antd";
import { withRouter } from "react-router-dom";
import {
  getEditingConditionFlag,
  getDraftPermitAmendmentForNOW,
} from "@common/selectors/permitSelectors";
import {
  setEditingConditionFlag,
  createPermitCondition,
  fetchPermitConditions,
  fetchDraftPermitByNOW,
  createStandardPermitCondition,
  fetchStandardPermitConditions,
} from "@common/actionCreators/permitActionCreator";
import CustomPropTypes from "@/customPropTypes";
import AddButton from "@/components/common/AddButton";
import Condition from "@/components/Forms/permits/conditions/Condition";
import NOWActionWrapper from "@/components/noticeOfWork/NOWActionWrapper";
import * as Permission from "@/constants/permissions";

const propTypes = {
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
  editingConditionFlag: PropTypes.bool.isRequired,
  setEditingConditionFlag: PropTypes.func.isRequired,
  createPermitCondition: PropTypes.func.isRequired,
  fetchPermitConditions: PropTypes.func.isRequired,
  fetchDraftPermitByNOW: PropTypes.func.isRequired,
  fetchStandardPermitConditions: PropTypes.func.isRequired,
  createStandardPermitCondition: PropTypes.func.isRequired,
  draftPermitAmendment: CustomPropTypes.permit.isRequired,
  match: PropTypes.shape({
    params: {
      type: PropTypes.string,
    },
  }).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }).isRequired,
};

const defaultProps = {};

export class AddCondition extends Component {
  state = { isEditing: false, conditionType: "SEC" };

  handleSubmit = (values) => {
    const isAdminDashboard = this.props.location.pathname.includes(
      "admin/dashboard/permit-conditions"
    );
    const payload = { ...values, condition_type_code: this.state.conditionType };
    if (isAdminDashboard) {
      return this.props
        .createStandardPermitCondition(this.props.match.params.type, payload)
        .then(() => {
          this.setState({ isEditing: false });
          this.props.fetchStandardPermitConditions(this.props.match.params.type);
          this.props.setEditingConditionFlag(false);
        });
    }
    return this.props
      .createPermitCondition(this.props.draftPermitAmendment.permit_amendment_guid, payload)
      .then(() => {
        this.setState({ isEditing: false });
        this.props.fetchPermitConditions(this.props.draftPermitAmendment.permit_amendment_guid);
        this.props.fetchDraftPermitByNOW(
          null,
          this.props.draftPermitAmendment.now_application_guid
        );
        this.props.setEditingConditionFlag(false);
      });
  };

  handleCancel = (value) => {
    this.props.setEditingConditionFlag(value);
    this.setState({ isEditing: false });
  };

  render = () => {
    const conditionMenu = () =>
      ({
        SEC: (
          <Menu>
            {(!this.props.initialValues.sibling_condition_type_code ||
              this.props.initialValues.sibling_condition_type_code === "SEC") && (
              <Menu.Item>
                <button
                  type="button"
                  className="full"
                  onClick={() => {
                    this.props.setEditingConditionFlag(true);
                    this.setState({ isEditing: true, conditionType: "SEC" });
                  }}
                >
                  Sub-Section
                </button>
              </Menu.Item>
            )}
            {(!this.props.initialValues.sibling_condition_type_code ||
              this.props.initialValues.sibling_condition_type_code === "CON") && (
              <Menu.Item>
                <button
                  type="button"
                  className="full"
                  onClick={() => {
                    this.props.setEditingConditionFlag(true);
                    this.setState({ isEditing: true, conditionType: "CON" });
                  }}
                >
                  Condition
                </button>
              </Menu.Item>
            )}
          </Menu>
        ),
        CON: (
          <Menu>
            <Menu.Item>
              <button
                type="button"
                className="full"
                onClick={() => {
                  this.props.setEditingConditionFlag(true);
                  this.setState({ isEditing: true, conditionType: "LIS" });
                }}
              >
                List Item
              </button>
            </Menu.Item>
          </Menu>
        ),
        LIS: (
          <Menu>
            <Menu.Item>
              <button
                type="button"
                className="full"
                onClick={() => {
                  this.props.setEditingConditionFlag(true);
                  this.setState({ isEditing: true, conditionType: "LIS" });
                }}
              >
                List Item
              </button>
            </Menu.Item>
          </Menu>
        ),
      }[this.props.initialValues.parent_condition_type_code]);

    const hasSibling = this.props.initialValues.sibling_condition_type_code;
    const addTitle = hasSibling ? "Add Another" : "Create";
    return (
      <>
        {!this.state.isEditing && this.props.editingConditionFlag && (
          <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
            <Dropdown
              className="full-height"
              overlay={conditionMenu()}
              placement="bottomLeft"
              disabled
            >
              <AddButton type="secondary" disabled>
                {addTitle}
              </AddButton>
            </Dropdown>
          </NOWActionWrapper>
        )}
        {!this.props.editingConditionFlag && (
          <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
            <Dropdown className="full-height" overlay={conditionMenu()} placement="bottomLeft">
              <AddButton type="secondary">{addTitle}</AddButton>
            </Dropdown>
          </NOWActionWrapper>
        )}
        {this.state.isEditing && (
          <Condition
            new
            handleCancel={() => this.handleCancel()}
            handleSubmit={(values) => this.handleSubmit(values)}
            initialValues={{
              ...this.props.initialValues,
              condition_type_code: this.state.conditionType,
            }}
          />
        )}
      </>
    );
  };
}

const mapStateToProps = (state) => ({
  editingConditionFlag: getEditingConditionFlag(state),
  draftPermitAmendment: getDraftPermitAmendmentForNOW(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      setEditingConditionFlag,
      fetchPermitConditions,
      createPermitCondition,
      fetchDraftPermitByNOW,
      createStandardPermitCondition,
      fetchStandardPermitConditions,
    },
    dispatch
  );

AddCondition.propTypes = propTypes;
AddCondition.defaultProps = defaultProps;

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AddCondition));
