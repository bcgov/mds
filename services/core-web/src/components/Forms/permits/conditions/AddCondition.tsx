import React, { FC, useState } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { isEmpty } from "lodash";
import { Menu, Dropdown } from "antd";
import { withRouter, RouteComponentProps } from "react-router-dom";
import {
  getEditingConditionFlag,
  getDraftPermitAmendmentForNOW,
} from "@mds/common/redux/selectors/permitSelectors";
import {
  setEditingConditionFlag,
  createPermitCondition,
  fetchPermitConditions,
  fetchDraftPermitByNOW,
  createStandardPermitCondition,
  fetchStandardPermitConditions,
} from "@mds/common/redux/actionCreators/permitActionCreator";
import AddButton from "@/components/common/buttons/AddButton";
import Condition from "@/components/Forms/permits/conditions/Condition";
import NOWActionWrapper from "@/components/noticeOfWork/NOWActionWrapper";
import * as Permission from "@/constants/permissions";
import { ActionCreator } from "@mds/common/interfaces/actionCreator"
import { IDraftPermitAmendment } from "@mds/common";

interface AddCondtionProps {
  initialValues: any;
  editingConditionFlag?: boolean;
  setEditingConditionFlag?: ActionCreator<typeof setEditingConditionFlag>;
  createPermitCondition?: ActionCreator<typeof createPermitCondition>;
  fetchPermitConditions?: ActionCreator<typeof fetchPermitConditions>;
  fetchDraftPermitByNOW?: ActionCreator<typeof fetchDraftPermitByNOW>;
  fetchStandardPermitConditions?: ActionCreator<typeof fetchStandardPermitConditions>;
  createStandardPermitCondition?: ActionCreator<typeof createStandardPermitCondition>;
  draftPermitAmendment?: IDraftPermitAmendment;
  match?: any;
  location?: any;
  layer: number;
}

const typeFromURL = {
  "sand-and-gravel": "SAG",
  quarry: "QCA",
  exploration: "MIN",
  placer: "PLA",
};

export const AddCondition: FC<RouteComponentProps & AddCondtionProps> = (props) => {
  const [isEditing, setIsEditing] = useState(false);
  const [conditionType, setConditionType] = useState("SEC");

  const handleSubmit = (values) => {
    const isDraftPermit = !isEmpty(props.draftPermitAmendment);
    const permitAmendmentGuid = isDraftPermit
      ? props.draftPermitAmendment.permit_amendment_guid
      : props.match.params.id;
    const isAdminDashboard = props.location.pathname.includes("admin/permit-condition-management");

    const payload = { ...values, condition_type_code: conditionType };
    if (isAdminDashboard) {
      return props
        .createStandardPermitCondition(typeFromURL[props.match.params.type], payload)
        .then(() => {
          setIsEditing(false);
          props.fetchStandardPermitConditions(typeFromURL[props.match.params.type]);
          props.setEditingConditionFlag(false);
        });
    }
    return props.createPermitCondition(permitAmendmentGuid, payload).then(() => {
      setIsEditing(false);
      props.fetchPermitConditions(permitAmendmentGuid);
      if (isDraftPermit) {
        props.fetchDraftPermitByNOW(null, props.draftPermitAmendment.now_application_guid);
      }
      props.setEditingConditionFlag(false);
    });
  };

  const handleCancel = (value) => {
    props.setEditingConditionFlag(value);
    setIsEditing(false);
  };

  const conditionMenu = () =>
    ({
      SEC: (
        <Menu>
          {(!props.initialValues.sibling_condition_type_code ||
            props.initialValues.sibling_condition_type_code === "SEC") && (
            <Menu.Item>
              <button
                type="button"
                className="full"
                onClick={() => {
                  props.setEditingConditionFlag(true);
                  setIsEditing(true);
                  setConditionType("SEC");
                }}
              >
                Sub-Section
              </button>
            </Menu.Item>
          )}
          {(!props.initialValues.sibling_condition_type_code ||
            props.initialValues.sibling_condition_type_code === "CON") && (
            <Menu.Item>
              <button
                type="button"
                className="full"
                onClick={() => {
                  props.setEditingConditionFlag(true);
                  setIsEditing(true);
                  setConditionType("CON");
                }}
              >
                Condition
              </button>
            </Menu.Item>
          )}
        </Menu>
      ),
      CON: (
        <Menu>
          <Menu.Item>
            <button
              type="button"
              className="full"
              onClick={() => {
                props.setEditingConditionFlag(true);
                setIsEditing(true);
                setConditionType("LIS");
              }}
            >
              List Item
            </button>
          </Menu.Item>
        </Menu>
      ),
      LIS: (
        <Menu>
          <Menu.Item>
            <button
              type="button"
              className="full"
              onClick={() => {
                props.setEditingConditionFlag(true);
                setIsEditing(true);
                setConditionType("LIS");
              }}
            >
              List Item
            </button>
          </Menu.Item>
        </Menu>
      ),
    }[props.initialValues.parent_condition_type_code]);

  const hasSibling = props.initialValues.sibling_condition_type_code;
  const addTitle = hasSibling ? "Add Another" : "Create";
  return (
    <>
      {!isEditing && props.editingConditionFlag && (
        <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
          {/* @ts-ignore */}
          <Dropdown
            className="full-height"
            overlay={conditionMenu()}
            placement="bottomLeft"
            disabled
          >
            {/* @ts-ignore */}
            <AddButton type="secondary" disabled>
              {addTitle}
            </AddButton>
          </Dropdown>
        </NOWActionWrapper>
      )}
      {!props.editingConditionFlag && (
        <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
          {/* @ts-ignore */}
          <Dropdown className="full-height" overlay={conditionMenu()} placement="bottomLeft">
            {/* @ts-ignore */}
            <AddButton type="secondary">{addTitle}</AddButton>
          </Dropdown>
        </NOWActionWrapper>
      )}
      {isEditing && (
        <Condition
          new
          handleCancel={() => handleCancel(false)}
          handleSubmit={(values) => handleSubmit(values)}
          initialValues={{
            ...props.initialValues,
            condition_type_code: conditionType,
          }}
        />
      )}
    </>
  );
};

const mapStateToProps = (state) => ({
  editingConditionFlag: getEditingConditionFlag(state),
  draftPermitAmendment: getDraftPermitAmendmentForNOW(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      setEditingConditionFlag,
      fetchPermitConditions,
      createPermitCondition,
      fetchDraftPermitByNOW,
      createStandardPermitCondition,
      fetchStandardPermitConditions,
    },
    dispatch
  );

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(AddCondition) as FC<
    AddCondtionProps & RouteComponentProps
  >
);
