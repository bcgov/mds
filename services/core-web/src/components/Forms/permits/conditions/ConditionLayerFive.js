import React, { useState } from "react";
import PropTypes from "prop-types";
import { Col, Row, Button } from "antd";
import { UpOutlined, DownOutlined } from "@ant-design/icons";
import Highlight from "react-highlighter";
import { highlightPermitConditionVariables } from "@common/utils/helpers";
import { TRASHCAN, EDIT_OUTLINE_VIOLET } from "@/constants/assets";
import NOWActionWrapper from "@/components/noticeOfWork/NOWActionWrapper";
import * as Permission from "@/constants/permissions";
import ConditionForm from "@/components/Forms/permits/conditions/ConditionForm";

const propTypes = {
  condition: PropTypes.objectOf(PropTypes.any),
  new: PropTypes.bool,
  handleSubmit: PropTypes.func,
  handleCancel: PropTypes.func,
  handleDelete: PropTypes.func,
  reorderConditions: PropTypes.func,
  setConditionEditingFlag: PropTypes.func,
  initialValues: PropTypes.objectOf(PropTypes.any),
  editingConditionFlag: PropTypes.bool.isRequired,
  isViewOnly: PropTypes.bool,
};

const defaultProps = {
  condition: undefined,
  new: false,
  handleSubmit: () => {},
  handleCancel: () => {},
  handleDelete: () => {},
  reorderConditions: () => {},
  setConditionEditingFlag: () => {},
  initialValues: {},
  isViewOnly: false,
};

const ConditionLayerFive = (props) => {
  // eslint-disable-next-line no-unused-vars
  const [isEditing, setIsEditing] = useState(props.new);
  return (
    <>
      <Row gutter={[8, 16]} className={isEditing || props.isViewOnly ? "" : "hover-row"}>
        {!isEditing && <Col span={4} />}
        <Col span={props.isViewOnly ? 2 : 1}>{!isEditing && props.condition.step}</Col>
        <Col span={props.isViewOnly ? 14 : 15}>
          {!isEditing && (
            <Highlight className="injectable-string" search={highlightPermitConditionVariables()}>
              {props.condition.condition}
            </Highlight>
          )}
        </Col>
        <Col span={3} className="float-right show-on-hover">
          {!isEditing && !props.isViewOnly && (
            <div className="float-right">
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => {
                    props.reorderConditions(props.condition, true);
                  }}
                  disabled={props.editingConditionFlag}
                >
                  <UpOutlined />
                </Button>
              </NOWActionWrapper>
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => {
                    props.reorderConditions(props.condition, false);
                  }}
                  disabled={props.editingConditionFlag}
                >
                  <DownOutlined />
                </Button>
              </NOWActionWrapper>
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => {
                    props.setConditionEditingFlag(true);
                    setIsEditing(!isEditing);
                  }}
                  disabled={props.editingConditionFlag}
                >
                  <img
                    className={props.editingConditionFlag ? "disabled-icon" : ""}
                    name="edit"
                    src={EDIT_OUTLINE_VIOLET}
                    alt="Edit Condition"
                  />
                </Button>
              </NOWActionWrapper>
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => props.handleDelete(props.condition)}
                  disabled={props.editingConditionFlag}
                >
                  <img
                    className={props.editingConditionFlag ? "disabled-icon" : ""}
                    name="remove"
                    src={TRASHCAN}
                    alt="Remove Condition"
                  />
                </Button>
              </NOWActionWrapper>
            </div>
          )}
        </Col>
      </Row>
      {isEditing && (
        <ConditionForm
          onCancel={() => {
            setIsEditing(!isEditing);
            props.setConditionEditingFlag(false);
            props.handleCancel(false);
          }}
          onSubmit={(values) => props.handleSubmit(values).then(() => setIsEditing(!isEditing))}
          initialValues={props.condition || props.initialValues}
          layer={5}
        />
      )}
    </>
  );
};

ConditionLayerFive.propTypes = propTypes;
ConditionLayerFive.defaultProps = defaultProps;

export default ConditionLayerFive;
