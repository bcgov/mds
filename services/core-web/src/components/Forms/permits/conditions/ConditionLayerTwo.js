import React, { useState } from "react";
import PropTypes from "prop-types";
import { Col, Row, Button } from "antd";
import { UpOutlined, DownOutlined } from "@ant-design/icons";
import { maxBy } from "lodash";
import { TRASHCAN, EDIT_OUTLINE_VIOLET } from "@/constants/assets";
import NOWActionWrapper from "@/components/noticeOfWork/NOWActionWrapper";
import * as Permission from "@/constants/permissions";
import ConditionLayerThree from "@/components/Forms/permits/conditions/ConditionLayerThree";
import AddCondition from "@/components/Forms/permits/conditions/AddCondition";
import ConditionForm from "@/components/Forms/permits/conditions/ConditionForm";

const propTypes = {
  condition: PropTypes.objectOf(PropTypes.any),
  new: PropTypes.bool,
  handleSubmit: PropTypes.func,
  handleCancel: PropTypes.func,
  handleDelete: PropTypes.func,
  reorderConditions: PropTypes.func,
  setConditionEditingFlag: PropTypes.func,
  initialValues: PropTypes.objectOf(PropTypes.any),
  editingConditionFlag: PropTypes.bool.isRequired,
  isViewOnly: PropTypes.bool,
};

const defaultProps = {
  condition: undefined,
  new: false,
  handleSubmit: () => {},
  handleCancel: () => {},
  handleDelete: () => {},
  reorderConditions: () => {},
  setConditionEditingFlag: () => {},
  initialValues: {},
  isViewOnly: false,
};

const ConditionLayerTwo = (props) => {
  // eslint-disable-next-line no-unused-vars
  const [isEditing, setIsEditing] = useState(props.new);
  return (
    <>
      <Row gutter={[8, 16]} className={isEditing || props.isViewOnly ? "" : "hover-row"}>
        {!isEditing && (
          <>
            <Col span={1} />
            <Col span={props.isViewOnly ? 2 : 1}>{!isEditing && props.condition.step}</Col>
          </>
        )}
        <Col
          className={props.condition.condition_type_code === "SEC" ? "field-title" : ""}
          span={props.isViewOnly ? 18 : 19}
        >
          {!isEditing && props.condition.condition}
        </Col>
        <Col span={3} className="float-right show-on-hover">
          {!isEditing && !props.isViewOnly && (
            <div className="float-right">
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => {
                    props.reorderConditions(props.condition, true);
                  }}
                  disabled={props.editingConditionFlag}
                >
                  <UpOutlined />
                </Button>
              </NOWActionWrapper>
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => {
                    props.reorderConditions(props.condition, false);
                  }}
                  disabled={props.editingConditionFlag}
                >
                  <DownOutlined />
                </Button>
              </NOWActionWrapper>
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => {
                    props.setConditionEditingFlag(true);
                    setIsEditing(!isEditing);
                  }}
                  disabled={props.editingConditionFlag}
                >
                  <img
                    className={props.editingConditionFlag ? "disabled-icon" : ""}
                    name="edit"
                    src={EDIT_OUTLINE_VIOLET}
                    alt="Edit Condition"
                  />
                </Button>
              </NOWActionWrapper>
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="DFT">
                <Button
                  ghost
                  className="no-margin"
                  size="small"
                  type="primary"
                  onClick={() => props.handleDelete(props.condition)}
                  disabled={props.editingConditionFlag}
                >
                  <img
                    className={props.editingConditionFlag ? "disabled-icon" : ""}
                    name="remove"
                    src={TRASHCAN}
                    alt="Remove Condition"
                  />
                </Button>
              </NOWActionWrapper>
            </div>
          )}
        </Col>
      </Row>
      {isEditing && (
        <ConditionForm
          onCancel={() => {
            setIsEditing(!isEditing);
            props.setConditionEditingFlag(false);
            props.handleCancel(false);
          }}
          onSubmit={(values) => props.handleSubmit(values).then(() => setIsEditing(!isEditing))}
          initialValues={props.condition || props.initialValues}
          layer={2}
        />
      )}
      {props.condition &&
        props.condition.sub_conditions.map((condition) => (
          <ConditionLayerThree
            condition={condition}
            reorderConditions={props.reorderConditions}
            handleSubmit={props.handleSubmit}
            handleDelete={props.handleDelete}
            setConditionEditingFlag={props.setConditionEditingFlag}
            editingConditionFlag={props.editingConditionFlag}
            isViewOnly={props.isViewOnly}
          />
        ))}
      {!isEditing && !props.isViewOnly && (
        <Row gutter={[8, 16]}>
          <Col span={22} offset={1}>
            <AddCondition
              initialValues={{
                condition_category_code: props.condition.condition_category_code,
                condition_type_code: "LIS",
                display_order:
                  props.condition.sub_conditions.length === 0
                    ? 1
                    : maxBy(props.condition.sub_conditions, "display_order").display_order + 1,
                parent_permit_condition_id: props.condition.permit_condition_id,
                permit_amendment_id: props.condition.permit_amendment_id,
                parent_condition_type_code: props.condition.condition_type_code,
                sibling_condition_type_code:
                  props.condition.sub_conditions.length === 0
                    ? null
                    : props.condition.sub_conditions[0].condition_type_code,
              }}
              layer={2}
            />
          </Col>
        </Row>
      )}
    </>
  );
};

ConditionLayerTwo.propTypes = propTypes;
ConditionLayerTwo.defaultProps = defaultProps;

export default ConditionLayerTwo;
