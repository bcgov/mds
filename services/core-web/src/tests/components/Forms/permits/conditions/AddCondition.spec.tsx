import React from "react";
import { shallow } from "enzyme";
import { AddCondition } from "@/components/Forms/permits/conditions/AddCondition";

const dispatchProps: any = {};
const props: any = {};

const setupDispatchProps = () => {
  dispatchProps.setEditingConditionFlag = jest.fn();
  dispatchProps.createPermitCondition = jest.fn();
  dispatchProps.fetchPermitConditions = jest.fn();
  dispatchProps.fetchDraftPermitByNOW = jest.fn();
  dispatchProps.fetchStandardPermitConditions = jest.fn();
  dispatchProps.createStandardPermitCondition = jest.fn();
};

const setupProps = () => {
  props.AddCondition = [];
  props.permitConditionCategoryOptions = [];
  props.editingConditionFlag = false;
  props.isNoWApplication = true;
  props.hasSourceAddCondition = true;
  props.draftPermitAmendment = {};
  props.initialValues = {};
  props.draftPermitAmendment = {};
  props.match = {};
  props.location = { pathname: "" };
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("AddCondition", () => {
  it("renders properly", () => {
    const component = shallow(<AddCondition {...dispatchProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
