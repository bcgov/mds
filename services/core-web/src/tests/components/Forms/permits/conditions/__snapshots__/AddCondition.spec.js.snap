// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCondition renders properly 1`] = `
<Fragment>
  <withRouter(Connect(NOWActionWrapper))
    permission="role_edit_permits"
    tab="DFT"
  >
    <Dropdown
      className="full-height"
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      placement="bottomLeft"
    >
      <AddButton
        type="secondary"
      >
        Create
      </AddButton>
    </Dropdown>
  </withRouter(Connect(NOWActionWrapper))>
</Fragment>
`;
